Standards/scs-xxxx-v1-ipv4-networking-standard.md

```diff
@@ -0,0 +1,79 @@
+---
+title: IPv4 Networking Standard
+type: Standard
+status: Proposal
+track: IaaS
+---
+
+## Introduction
+
+This document outlines the standardized approach for the management and allocation of
+public IPv4 addresses within Sovereign Cloud Stack (SCS) environments. Its aim is to
+ensure a consistent, secure, and efficient methodology for IP address provisioning
+across all SCS cloud services.
+
+## Motivation
+
+The motivation behind establishing this standard is to enhance interoperability, improve
+security measures, and streamline the operational processes across different SCS clouds.
+It addresses the need for a unified procedure in handling IPv4 networking to facilitate
+ease of use for IaaS users and DevOps teams.
+
+## Design Considerations
+
+### Options considered
+
+#### _Neutron Routers_
+
+Usage of Neutron Routers: To manage traffic between internal and external networks Neutron Routers **MUST** be used as the default gateway for VMs requiring access to external networks and the internet, thereby facilitating the routing of traffic and enhancing network security.
+
+CSPs **SHOULD** use OVN or L3agent as High Availability (HA) service deployments.
+Standard external networks **MUST NOT** be made accessible as _shared networks_. It is advised that external networks are only reachable by the usage of routing and floating IPs.
+However, for special use cases like certain storage or VPN solutions it could be useful to allow _direct access networks_.
+
+External networks and subnets **SHOULD** (very strong should) be configured with _--no-dhcp_ (DHCP - Dynamic Host Configuration Protocol). It is more secure to configure it like this, since it gives less space for reflection attacks, e.g. _Denial of Service_ (DOS) attacks. If _dhcp_ is configured, certain firewall configurations **MUST** be made to catch IPs from the _Neutron dhcp agent_ in the public network.
+
+#### _Neutron Plugins_
+
+Neutron Plugins: A SCS conform CSP **MAY** use RBAC and VPNaaS plugins.
+(Neutron RBAC needs to be configured explicitly to be able to use it. If configured, Neutron configurations can be shared across OpenStack projects. It also can be beneficial for admins, since an admin could bind external networks only to certain projects.)
```

Comment:
I still wonder which RBAC do you mean? Can you point me towards this please?

Because it is already confusing to differ between the keystone roles and scopes RBAC that has to be implemented by every OpenStack service and this `network rbac` extension of Neutron, which allows sharing of resources - that I don't recommend. Maybe we need a standard that forbids the usage and adjusts the policy?

Reply:
Neutrons RBAC extension allows sharing of various resources among projects and is the mechanism that neutron uses to implement provider networks, so it's not really an optional feature. Rules can be created with the actions `access_as_shared` and `access_as_external` and a target_project_id that can be either an actual project id or `*` for sharing with all projects.
To my knowledge it does not require any explicit extra configuration, as it is part of the ML2 core plugin.

Neutrons default policies allow only the owner of a resource to share it, and allows only admin users to create rules with the wildcard `*` target.
This would still allow a malicius non-admin user to create a faux external network and offer it to selected target projects for whom it might not be immediately obvious that this network was not provided by the CSP (`openstack network list` does not even show the project id of the networks per default).
This may give the attacker the opportunity to intercept external traffic of the target project.
Preventing non-admin users from sharing networks is at least something we should consider, especially if we do not have a naming convention.